src/pydot/classes.py

```diff
@@ -0,0 +1,83 @@
+# SPDX-FileCopyrightText: 2024 pydot contributors
+#
+# SPDX-License-Identifier: MIT
+
+"""Frozen dictionaries."""
+
+import copy
+
+
+class FrozenDict(dict):
+    """Frozen dictionary, values are immutable after creation.
+
+    Extended version of ASPN's Python Cookbook Recipe:
+    https://code.activestate.com/recipes/414283/
+
+    This version freezes dictionaries used as values within dictionaries."""
+
+    _block_msg = "A frozendict cannot be modified."
```

Comment:
I think I hate myself a little bit, for not fixing the "FrozenDict" capitalization here. :roll_eyes: